piksi_tools/console/settings_view.py

```diff
@@ -712,203 +742,56 @@ def settings_display_setup(self, do_read_finished=True):
-            b'\0')[:4]
-        section, setting, value, format_type = (section.decode(KEY_ENCODING),
-                                                setting.decode(KEY_ENCODING),
-                                                value.decode(VALUE_ENCODING),
-                                                format_type.decode(VALUE_ENCODING))
-        self.ordering_counter += 1
-        if format_type == '':
-            format_type = None
-        else:
-            setting_type, setting_format = format_type.split(':')
-        if section not in self.settings:
-            self.settings[section] = {}
-        # setting exists, we won't reinitilize it but rather update existing setting
-        dict_setting = self.settings[section].get(setting, False)
-        if dict_setting:
-            dict_setting._prevent_revert_thread = True
-            dict_setting.value = value
-            dict_setting._prevent_revert_thread = False
-            dict_setting.ordering = self.ordering_counter
-            if format_type is not None and setting_type == 'enum':
-                enum_values = setting_format.split(',')
-                dict_setting.enum_values = enum_values
-        else:
-            if format_type is None:
-                # Plain old setting, no format information
-                self.settings[section][setting] = Setting(
-                    setting,
-                    section,
-                    value,
-                    ordering=self.ordering_counter,
-                    settings=self)
-            else:
-                if setting_type == 'enum':
-                    enum_values = setting_format.split(',')
-                    self.settings[section][setting] = EnumSetting(
-                        setting,
-                        section,
-                        value,
-                        enum_values,
-                        ordering=self.ordering_counter,
-                        settings=self)
-                else:
-                    # Unknown type, just treat is as a string
-                    self.settings[section][setting] = Setting(
-                        setting, section, value, settings=self, ordering=self.ordering_counter)
-        # remove index from list of pending items
-        if sbp_msg.index in self.pending_settings:
-            self.pending_settings.remove(sbp_msg.index)
-        if len(self.pending_settings) == 0:
-            self.pending_settings = list(range(self.enumindex, self.enumindex + BATCH_WINDOW))
-            self.enumindex += BATCH_WINDOW
-            self._send_pending_settings_by_index()
-            self._restart_retry_thread()
+        if finish:
+            self.finish_read()
 
     def piksi_startup_callback(self, sbp_msg, **metadata):
-        self._settings_read_by_index()
-
-    def set(self, section, name, value):
-        self.link(
-            MsgSettingsWrite(setting=b'%s\0%s\0%s\0' % (section, name, value)))
+        self._settings_read_all()
```

Comment:
Perhaps we can call a read_all if we are getting a never before seen SBP Sender_ID as well to avoid the console's local cache not matching the device that is plugged in.

Reply:
I presume this would be a new feature? If so I propose to do it in a separate PR.

Reply:
yes this was just me brainstorming; no action required for this PR.